src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/RazorProjectInfoFileSerializer.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using MessagePack;
+using MessagePack.Resolvers;
+using Microsoft.AspNetCore.Razor.ProjectSystem;
+using Microsoft.AspNetCore.Razor.Serialization.MessagePack.Resolvers;
+
+namespace Microsoft.CodeAnalysis.Razor.Serialization;
+
+internal class RazorProjectInfoFileSerializer : IRazorProjectInfoFileSerializer
```

Comment:
>RazorProjectInfoFileSerializer [](http://example.com/codeflow?start=15&length=30)

I agree with David that protecting against file I/O exceptions would be nice. I don't know if it would be better to protect against a single failure, log, and proceed with the rest, or fail the entire batch and possibly retry?